assets/js/components/settings/SettingsRenderer.js

```diff
@@ -25,18 +25,21 @@ import { useEffect } from '@wordpress/element';
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
-import SettingsEdit from './SettingsEdit';
-import SettingsView from './SettingsView';
-import { STORE_NAME } from '../../datastore/constants';
+import { STORE_NAME as CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
 const { useSelect, useDispatch } = Data;
+const NullComponent = () => null;
+
+export default function SettingsRenderer( { slug, isOpen, isEditing } ) {
+	const storeName = `modules/${ slug }`;
+	const isDoingSubmitChanges = useSelect( ( select ) => select( storeName )?.isDoingSubmitChanges?.() );
+	const haveSettingsChanged = useSelect( ( select ) => select( storeName )?.haveSettingsChanged?.() );
+	const SettingsEdit = useSelect( ( select ) => select( CORE_MODULES ).getModule( slug )?.settingsEditComponent ) || NullComponent;
+	const SettingsView = useSelect( ( select ) => select( CORE_MODULES ).getModule( slug )?.settingsViewComponent ) || NullComponent;
 
-export default function SettingsMain( { isOpen, isEditing } ) {
-	const isDoingSubmitChanges = useSelect( ( select ) => select( STORE_NAME ).isDoingSubmitChanges() );
-	const haveSettingsChanged = useSelect( ( select ) => select( STORE_NAME ).haveSettingsChanged() );
 	// Rollback any temporary selections to saved values if settings have changed and no longer editing.
-	const { rollbackSettings } = useDispatch( STORE_NAME );
+	const { rollbackSettings } = useDispatch( storeName ) || {};
 	useEffect( () => {
-		if ( haveSettingsChanged && ! isDoingSubmitChanges && ! isEditing ) {
+		if ( rollbackSettings && haveSettingsChanged && ! isDoingSubmitChanges && ! isEditing ) {
 			rollbackSettings();
 		}
 	}, [ haveSettingsChanged, isDoingSubmitChanges, isEditing ] );
```

Comment:
I think generally, yes but this is now it was before. I only added the check to make sure the action exists before calling it. I'll add it to the array though since there should be no harm in that, but in this case it shouldn't matter.